.github/workflows/optimize-unit-tests.yml

```diff
@@ -49,12 +49,25 @@ jobs:
         retention-days: 7
         if-no-files-found: ignore
 
+    - name: Observe build status
+      if: always()
+      continue-on-error: true
+      uses: ./.github/actions/observe-build-status
+      with:
+        build_status: ${{ job.status }}
+        secret_vault_secretId: ${{ secrets.VAULT_SECRET_ID }}
+        secret_vault_address: ${{ secrets.VAULT_ADDR }}
+        secret_vault_roleId: ${{ secrets.VAULT_ROLE_ID }}
+
   unit-tests-frontend:
     name: Optimize Unit Tests - Frontend
     runs-on: ubuntu-latest
-    timeout-minutes: 30
+    timeout-minutes: 10
     needs: detect-changes
     if: ${{ needs.detect-changes.outputs.frontend-changes == 'true' }}
```

Comment:
Can this change detection logic here be refactored into the common place in `.github/actions/paths-filter` to avoid 2 layers of change detection where one would be enough?

Reply:
You mean just moved from the `optimzie-paths-filter` to `paths-filter`?

Reply:
I also meant to not execute `./.github/actions/paths-filter` twice (once in `ci.yml` and once in this file) since that is redundant and thus unexpected.